### PR TITLE
 Implement Intl.ListFormat

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: src/runtime/StaticStrings.h [hunk 1]
```diff
@@ -310,6 +310,7 @@ namespace Escargot {
     F(length)                     \
     F(let)                        \
     F(link)                       \
+    F(ListFormat)                 \
```
Comment (clover2123) [inline src/runtime/StaticStrings.h:313]: (tiny fix) Word starts with upper case is located above according to the sorting rule.

File: src/intl/IntlListFormat.cpp [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+#include "runtime/ArrayObject.h"
+#include "runtime/ErrorObject.h"
+#include "runtime/IteratorObject.h"
+#include "Intl.h"
+#include "IntlListFormat.h"
+
+namespace Escargot {
+
+static const char* const intlListFormatRelevantExtensionKeys[0] = {};
+static size_t intlListFormatRelevantExtensionKeysLength = 0;
```
Comment (clover2123) [inline src/intl/IntlListFormat.cpp:35]: It seems that `intlListFormatRelevantExtensionKeys` is used only once in this file.
Is there any reason that it should be allocated in the global scope?

File: src/runtime/StaticStrings.h (hunk 1 above)
Comment (ksh8281) [reply src/runtime/StaticStrings.h:313]: Okay!

File: src/intl/IntlListFormat.cpp (hunk 2 above)
Comment (ksh8281) [reply src/intl/IntlListFormat.cpp:35]: What a nice catch! I will fix it!

**Review by clover2123 — APPROVED** [review]
LGTM

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**